Sources/ConfettiView/ConfettiView.swift

```diff
@@ -30,7 +33,10 @@ public final class ConfettiView: UIView {
                     3.0 seconds by default.
     */
     public func emit(with contents: [Content],
-                     for duration: TimeInterval = 3.0) {
+                     for duration: TimeInterval = 3.0,
+                     completion: ((ConfettiView) -> Void)? = nil) {
+        
+        self.completion = completion
```

Comment:
Is there any reason for storing `completion` as a property? At first glance, the code would work just the same without this line (and removing the `self.` on 164).

Reply:
Also, calling `emit` twice consecutively would wipe out the completion block from the first call, no?

Reply:
You're right. My use case was to emit one time and do some after work completion. Forget the idea.

Reply:
@chepiok No no, I think it's a quite a good idea. It's just that the nature of Core Animation APIs makes reentrancy difficult.

After checking out your PR branch, I realized why you added the property — you need a reference for the `animationDidStop` callback. That makes sense. I was confused by the partial diff view, and conflated that with the completion block of the transaction that added the animation.

I agree that it'd be helpful to be notified about the completion of the confetti animation. But the only way I see for this to work would be to set that handler as a property on `ConfettiView` instead of a parameter to `emit` (I don't believe you can set the handler as a property on the `CALayer`, which would be the other way to pass the state through to the `animationDidStop` callback). What do you think about that approach?

Reply:
Changing the property `completion` to public and having `completion` parameter optional on `emit` allows the user to use its best solution. Up to you to choose what is the most useful and elegant.

And yes in term of logic the best place is on the `CALayer` but I don't think it's possible.

Reply:
@chepiok A possible solution to this just randomly occurred to me. What if we subclass `CAKeyframeAnimation` and add `completion` as a property? Then in `animationDidStop`, we could do something like this:

```swift
 public func animationDidStop(_ animation: CAAnimation, finished flag: Bool) {
    if let layer = animation.value(forKey: kAnimationLayerKey) as? CALayer {
        layer.removeAllAnimations()
        layer.removeFromSuperlayer()
        self.completion?(self)
    }

    if let animation = animation as? Animation /* i.e. ConfettiView.Animation */ {
        animation.completion?()
    }
}
```

Would you be interested in / available to rewriting your PR following that approach?